pkg/ccl/backupccl/backup_test.go

```diff
@@ -2325,6 +2325,67 @@ INSERT INTO sc.tb2 VALUES ('hello');
 		sqlDB.ExpectErr(t, `pq: schema "unused" already exists`, `USE d; CREATE SCHEMA unused`)
 	})
 
+	// Tests backing up and restoring all tables in requested user defined
+	// schemas.
+	t.Run("all-tables-in-requested-schema", func(t *testing.T) {
+		_, _, sqlDB, _, cleanupFn := BackupRestoreTestSetup(t, singleNode, 0, InitManualReplication)
+		defer cleanupFn()
+
+		sqlDB.Exec(t, `
+CREATE TABLE table_in_data (x INT);
+
+CREATE DATABASE foo;
+USE foo;
+CREATE SCHEMA schema_in_foo;
+CREATE TABLE schema_in_foo.tb1 (x INT);
+
+CREATE SCHEMA schema_in_foo2;
+CREATE TABLE schema_in_foo2.tb1 (x INT);
+
+CREATE TABLE tb2 (y INT);
+`)
+
+		for _, tc := range []struct {
+			name                   string
+			target                 string
+			expectedTablesInBackup [][]string
+		}{
+			{
+				name:                   "fully-qualified-target",
+				target:                 "foo.schema_in_foo.*",
+				expectedTablesInBackup: [][]string{{"schema_in_foo", "tb1"}},
+			},
+			{
+				name:                   "schema-qualified-target",
+				target:                 "schema_in_foo.*",
```

Comment:
Is it legal for a database and schema to have the same name? Is it ambiguous what this means in that case?

Reply:
It's legal but I don't think we have any databases and schemas with the same name here?

Reply:
Sorry I misunderstood your question. In that case it should behave in the same way. Given database `foo` and schema `foo.foo`, if the current database in the SQL session is set to `foo`, the target `foo.*` would include all tables within schema `foo.foo`, this is because the resolver would try to look up the schema name in the current db first. If we're in a different db that does not have schema `foo`, `foo.*` would back up all tables in db `foo`.

Reply:
Thanks for clarifying, that sounds reasonable. I think it would be good to add a test case covering this case so it's explicit that's what's expected.

Let's also update the release note to mention this, or alternatively we could file a separate docs issue to call this out in the backup docs. Perhaps we should consider this more of a SQL change rather than a bug fix. Perhaps something like:
```
(sql change): BACKUP now supports backing up tables in a specified schema (e.g. BACKUP my_schema.*). Schemas will be resolved before databases, so `my_object.*` will resolve to a schema of that name in the current database before matching a database with that name.
```